code/modules/mob/living/carbon/xenomorph/xenoprocs.dm

```diff
@@ -254,6 +254,8 @@
 			if(!hive?.living_xeno_ruler || hive.living_xeno_ruler.loc.z == loc.z)
 				if(upgrade_stored >= xeno_caste.upgrade_threshold)
 					if(health == maxHealth && !incapacitated() && !handcuffed)
+						if (agility || fortify || crest_defense) // just return until dude stop using this, avoids breaking armor values and causing issues :)
```

Comment:
signal this

Reply:
to be clear: don't block upgrade because you're using an ability, make it send a signal for active abilities to upgrade their bonuses without interupting the ability

Reply:
i'll need help with this if possible.